pkg/endpoint/endpoint.go

```diff
@@ -1673,9 +1682,21 @@ func (e *Endpoint) RunMetadataResolver(bwm bandwidth.Manager, resolveMetadata Me
 
 				pod, cp, identityLabels, info, _, err := resolveMetadata(ns, podName)
 				if err != nil {
+					// If we were unable to fetch the k8s endpoints then
+					// we will mark the endpoint with the init identity.
+					if initialLabels != nil {
```

Comment:
godoc comment says `If this list is empty or nil` but we only check for `nil` here. Any reason we're not using `len(initialLabels) != 0` here?

Reply:
I see. Honestly, I find this a bit fragile. Have you considered just introducing a boolean for that?